### PR TITLE
Access with malformed authorization header will return error code

### PR DESCRIPTION
**Pull request by Jackson-Wang-7**
### What changes are proposed in this pull request?

access with malformed authorization header will return error code `AuthorizationHeaderMalformed `.

### Why are the changes needed?

Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, describe the bug.

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
  1. change in user-facing APIs
  2. addition or removal of property keys
  3. webui

### CODE REVIEW

**Comment by Jackson-Wang-7** [comment]
image

File: core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java [hunk 1]
```diff
@@ -200,7 +201,7 @@ private String getUserFromSignature() throws S3Exception {
    * @return the user
    */
   @VisibleForTesting
-  public static String getUserFromAuthorization(String authorization) {
+  public static String getUserFromAuthorization(String authorization) throws S3Exception {
     if (authorization == null) {
       return null;
     }
```
Comment (ZhuTopher) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java:207]: After some discussion I agree that the "Authorization" header shouldn't be optional, so we can return the same `AuthorizationHeaderMalformed` error code here.

File: core/server/proxy/src/main/java/alluxio/proxy/s3/CompleteMultipartUploadHandler.java [hunk 2]
```diff
@@ -87,6 +87,9 @@ public void handle(String s, Request request, HttpServletRequest httpServletRequ
       if (!s.startsWith(mS3Prefix)) {
         return;
       }
+      if (!request.getMethod().equals("POST") || request.getParameter("uploadId") == null) {
+        return;
+      } // Otherwise, handle CompleteMultipartUpload
```
Comment (ZhuTopher) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/CompleteMultipartUploadHandler.java:92]: Please undo this change, at the moment we're relying on those logs for the _other_ S3 API actions too. If you want to, it'd be great to separate out these logs to a separate handler entirely.

**Review by ZhuTopher — CHANGES_REQUESTED** [review]
Could you add some unit tests for this?
- AWS sigv2 should be rejected
- empty "Authorization" header should be rejected
- AWS sigv4 test: "user1" should not be able to see buckets owned by "user2"

File: core/server/proxy/src/main/java/alluxio/proxy/s3/CompleteMultipartUploadHandler.java (hunk 2 above)
Comment (Jackson-Wang-7) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/CompleteMultipartUploadHandler.java:92]: I do find that other S3 API will execute this code part. I'm just curious why it was designed like this. especially it's  `MultipartUploadHandler.handle()`. we may need a more common part (maybe a handler or filter?) to do this logging thing

File: core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java (hunk 1 above)
Comment (Jackson-Wang-7) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java]: done

**Comment by Jackson-Wang-7** [comment]
image

AWS sigv4 test: "user1" should not be able to see buckets owned by "user2"

**Comment by ZhuTopher** [comment]
Consistent test failure in `S3ClientRestApiTest` for `listAllMyBuckets()`; I'm merging into the targeted feature-branch and will address in that PR separately.

**Comment by ZhuTopher** [comment]
alluxio-bot, merge this please

**Comment by ZhuTopher** [comment]
alluxio-bot, force merge this please
